api/base/serializers.py

```diff
@@ -256,6 +269,18 @@ def many_init(cls, *args, **kwargs):
         kwargs['child'] = cls()
         return JSONAPIListSerializer(*args, **kwargs)
 
+    def validate_type(self, value):
```

Comment:
Now that we have `IDField` and `TypeField`, it'd be more cohesive to put this validation on the Fields themselves.

```python
class TypeField(ser.CharField):

     def to_internal_value(self, data):
         self.root.Meta.type_ != value:
             raise Conflict
          return super(....)..
```